File: ps/woman/0327/frog.py
N = int(input())
array = []
for _ in range(N):
    array.append(int(input()))
    
array.sort()

def get(target):
    low = 0;
    high = len(array)
    while low < high:
        mid = (low+high)//2
        if array[mid] >= target:
            high = mid
        else:
            low = mid+1

    return high;


result = 0;
for i in range(len(array)-1):
    for j in range(i+1, len(array)):
        jump = array[j] - array[i]
        result += get(array[j]+jump*2+1) - get(array[j]+jump)

print(result)
        

File: ps/woman/0327/frog.cpp
#include "bits/stdc++.h"
using namespace std;

int N;
int arr[1001] = {};
int search(int target) {
    int x = -1;
    for (int i = N; i >= 1; i/=2) {
        while (x+i < N && arr[x+i]<target) x+= i;

    }     
    return x;
}

int main() {
    scanf("%d", &N);
    for (int i = 0; i < N; i++) {
        scanf("%d", &arr[i]); 
    }

    int result = 0; 
    sort(arr, arr+N);
    for (int i = 0; i < N-1; i++) {
        for (int j = i+1; j < N; j++) {
            int jump = arr[j]-arr[i];
            result += search(arr[j]+2*jump+1) - search(arr[j]+jump);

        }
    }
    cout << result;

    return 0;
}
